core/src/main/java/io/questdb/Telemetry.java

```diff
@@ -103,6 +116,60 @@ public void consumeAll() {
         }
     }
 
+    public long getDirSize(FilesFacade ff, Path path, boolean topLevel, StringSink sink) {
+        if (topLevel) {
+            startDbSizeEstimateTimestamp = clock.getTicks();
+        }
+
+        long pFind = ff.findFirst(path.$());
+        long totalSize = 0L;
+
+        if (pFind > 0L) {
+            int len = path.size();
+            try {
+                do {
+                    if (clock.getTicks() - startDbSizeEstimateTimestamp > maxDbSizeEstimateTime) {
+                        break;
+                    }
+
+                    long nameUtf8Ptr = ff.findName(pFind);
+                    path.trimTo(len).concat(nameUtf8Ptr).$();
+                    if (ff.findType(pFind) == Files.DT_FILE) {
```

Comment:
Can `ff.findType()` return `Files.UNKNOWN`?

It can also return `Files.LINK`, but I guess that is ok. We can assume that links are directories.

Reply:
`Files.UNKNOWN` is associated with `.` and `..`.
The code checked for them on the else branch, so all good.